source/fundamentals/data-formats/pojo-customization.txt

```diff
@@ -10,78 +10,653 @@ POJO Customization
+
+- Use the ``@BsonIgnore`` annotation for a property to always skip
+  serialization. Make sure to enable annotations using the appropriate
+  :ref:`Conventions <conventions>`.
+- Create a custom class that overrides the ``shouldSerialize()`` method of the
+  ``PropertySerialization`` interface. Specify your custom implementation to
+  the ``PropertyModelBuilder`` which is accessible from the
+  ``ClassModelBuilder``.
+
+For more information on how to use the ``@BsonIgnore`` annotation in a POJO,
+see the section of this guide on :ref:`Annotations <annotations>`.
+
+The following sample code shows a custom class that implements the
+``PropertySerialization`` interface to override the default conditions
+by which to determine whether to serialize a field:
 
-- Description
-  - Define term: discriminator
-- Example
-  - Show each way (annotation and ClassModelBuilder method) to enable discriminator
-  
+.. code-block:: java
+   :emphasize-lines: 4
 
-Supporting POJOs without Default Constructors
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   public class CourteousAgeSerialization implements PropertySerialization<Integer> {
+       @Override
+       public boolean shouldSerialize(Integer value) {
+           return (value < 30);
+       }
+   }
 
-- Description, explaining when this is used
-- Example of BsonCreator and BsonProperty annotations to specify the constructor and parameter names to use
+The class above specifies that any integer above 29 is not serialized and
+therefore, not included in the MongoDB document. Suppose you applied this
+custom serialization behavior to the following sample POJO:
 
-Customizing Serialization
-~~~~~~~~~~~~~~~~~~~~~~~~~
+.. code-block:: java
 
-- Description, when to use
-- Examples
-  - Annotation
-  - PropertySerialization in PropertyModelBuilder
+   public class BirthdayInvitation {
+       private String name;
+       private Integer age;
+       private LocalDateTime eventDateTime;
 
+       // ...
+   }
+
+You can specify the custom serialization by adding the
+``CourteousAgeSerialization`` instance to the ``PropertyModelBuilder`` from
+the ``ClassModel`` property associated with the ``age`` field using the
+following code:
+
+.. code-block:: java
+
+   ClassModelBuilder<BirthdayInvitation> classModel = ClassModel.builder(BirthdayInvitation.class);
+   ((PropertyModelBuilder<Integer>) classModel.getProperty("age"))
+           .propertySerialization(new CourteousAgeSerialization());
+
+   PojoCodecProvider pojoCodecProvider = PojoCodecProvider.builder().register(classModel.build()).build();
+   CodecRegistry pojoCodecRegistry = fromRegistries(getDefaultCodecRegistry(), fromProviders(pojoCodecProvider));
+
+If you insert a POJO that contains a value greater than 29 in the ``age``
+field, the serialized document omits it. The resulting document should
+resemble the following:
+
+.. code-block:: json
+   :copyable: false
+
+   { "_id" : ObjectId("..."), "eventDateTime" : ..., "name" : "Galadriel" }
```

Comment:
This example is both education and hilarious. I would suggest introducing a sample full Galadriel document (including an age) before we show this resulting document, though, in case users want to confirm their understanding of the example.